src/ProjectTemplates/README.md

```diff
@@ -1,23 +1,48 @@
 # Templates
 
-## Getting Started
 These are project templates which are used in .NET Core for creating ASP.NET Core applications.
 
-## Prerequisites
+## Description
+
+The following contains a description of each sub-directory in the `ProjectTemplates` directory.
+
+- `BlazorTemplates.Tests`: Contains the source files for the Blazor template tests, these are currently split out due to not being Helix ready yet.
+- `Shared`: Contains a collection of shared constants and helper methods/classes including the infrastructure for managing dotnet processes to create, build, run template tests.
+- `Web.Client.ItemTemplates`: Contains the Web Client-Side File templates, includes things like less, scss, and typescript
+- `Web.ItemTemplates`: Contains the Web File templates, includes things like: protobuf, razor component, razor page, view import and start pages
+- `Web.ProjectTemplates`: Contains the ASP.NET Core Web Template pack, including Blazor Server, WASM, Empty, Grpc, Razor Class Library, RazorPages, MVC, WebApi.
+- `Web.Spa.ProjectTemplates`: Contains the Single Page Application templates for ASP.NET Core, including Anuglar, React, ReactRedux.
+- `migrations`: Contains migration related scripts.
+- `scripts`: Contains a collection of scripts that help running tests locally that avoid having to install the templates to the machine.
+- `test`: Contains the end to end template tests.
+- `testassets`: Contains assets used by the tests, like a dotnet tools installer
+
+### Build
+
 Some projects in this repository (like SignalR Java Client) require JDK installation and configuration of `JAVA_HOME` environment variable.
 1. If you don't have the JDK installed, you can find it from https://www.oracle.com/technetwork/java/javase/downloads/index.html
 1. After installation define a new environment variable named `JAVA_HOME` pointing to the root of the latest JDK installation (for Windows it will be something like `c:\Program Files\Java\jdk-12`).
 1. Add the `%JAVA_HOME%\bin` directory to the `PATH` environment variable
 
-## Building Templates
-1. Run `. .\activate.ps1` if you haven't already.
+To build the ProjectTemplates:
 
+1. Run `. .\activate.ps1` if you haven't already.
```

Comment:
We can remove these first few steps since they are applicable to entire repo and we already document them in the BuildFromSource.md.